pkg/dispers/dispers/requests.go

```diff
@@ -202,8 +203,44 @@ type Token struct {
 }
 
 // Instance describes the location of an instance and the token it had created
+// When Target received twice the same Instance, it needs to be able to consider the more recent item
 type Instance struct {
-	Host   string `json:"host,omitempty"`
-	Domain string `json:"domain,omitempty"`
-	Token  Token  `json:"token,omitempty"`
+	Domain           string    `json:"domain"`
+	SubscriptionDate time.Time `json:"date"`
+	Token            Token     `json:"token"`
+}
+  
+/*
+*
+Targets' Input & Output
+*
+*/
+
+// InputT contains information received by Target's enclave
+type InputT struct {
+	IsEncrypted         bool       `json:"isencrypted,omitempty"`
+	LocalQuery          LocalQuery `json:"localquery,omitempty"`
+	Targets             []string   `json:"Addresses,omitempty"`
+	EncryptedLocalQuery []byte     `json:"enc_localquery,omitempty"`
+	EncryptedTargets    []byte     `json:"enc_addresses,omitempty"`
```

Comment:
I just realized this is done for all others struct, but I don't think it's a good idea to have each parameter replicated in an encrypted version. That being said, let's keep it as it is for now, but we would probably change this in the future when we introduce encryption.

Reply:
It does not bother me to change it now if you want me to !